src/6_branch.js

```diff
@@ -1745,3 +1745,74 @@ Branch.prototype['closeBanner'] = wrap(0, function(done) {
 	});
 	done();
 });
+
+/**
+ * @function Branch.autoAppIndex
+ *
+ * @param {string} androidPackageName - _optional_ - Android App's package name
+ * @param {string} androidURL - _optional_ - A custom scheme for your Android App such as: 'example/home/cupertino/12345' where 'example' is the App's URI scheme and 'home/cupertino/12345' routes to unique content the App
+ * @param {string} iosAppStoreId - _optional_ - iTunes App Store ID for your iOS App 
+ * @param {string} iosURL - _optional_ - A custom scheme for your iOS App such as: 'example/home/cupertino/12345'
+ * @param {function(?Error)=} callback - _optional_ - Returns an error if unsuccessful
+ *
+ * This function generates and inserts Firebase App Indexing tags between the <head></head> section of your webpage. 
+ * Once inserted, these tags will help Google to index and surface content from your App in Google Search.
+ * 
+ * Resultant Firebase App Indexing tags will have the following format:
+ *
+ * <link rel="alternate" href="androidapp://{androidPackageName}/{androidURL}?{branch_tracking_params}"/>
+ * <link rel="alternate" href="ios-app://{iosAppStoreId}/{iosURL}?{branch_tracking_params}"/>
+ *
+ * Note: If optional parameters above are not specified then Branch will try to build App Indexing tags using your webpage's App Links tags.
+ * Also, if optional parameters are specified but App Indexing tags are already present on the webpage then, Branch will ignore what is passed into .autoAppIndex() 
+ * and append Branch tracking params to the end of the existing tags.
+ *
+ * Analytics related to Google's attempts to index your App via this method can be found from Source Analytics in Dashboard where channel is set to 'Firebase-App-Indexing'.
+ *
+ * ##### Usage
+ * ```js
+ * branch.autoAppIndex(
+ *     data,
+ *     callback (err)
+ * );
+ *```
+ * ##### Example
+ * ```js
+ * branch.autoAppIndex({ 
+ *     iosAppId:'123456789',
+ *     iosURL:'example/home/cupertino/12345',
+ *     androidPackageName:'com.somecompany.app',
+ *     androidURL:'example/home/cupertino/12345'
+ * }, function(err) { console.log(err); });
+ *```
+ *
+ * ##### Callback Format
+ * ```js
+ * callback(
+ *      "Error message"
+ * );
+ * ```
+ * ___
+ */
+/*** +TOC_HEADING &Firebase App Indexing& ^ALL ***/
+/*** +TOC_ITEM #autoAppIndex-callback &.autoAppIndex()& ^ALL ***/
+Branch.prototype['autoAppIndex'] = wrap(callback_params.CALLBACK_ERR, function(done, options) {
+	var self = this;
+	options = options || {};
+	appindexing.checkForAppIndexingTags();
+
+	if (appindexing.state['iOSAppIndexingTagsPresent'] === false || appindexing.state['androidAppIndexingTagsPresent'] ===  false) {
+		appindexing.checkOptions(options);
```

Comment:
On second thought, this would require you to separately check the state of the config to set `appindexing.state['androidDetailsComplete']` and `appindexing.state['iOSDetailsComplete']` so perhaps it's fine for now. Cancel this suggestion.